test/abstractarray.jl

```diff
@@ -80,7 +80,7 @@ using StaticArrays, Test, LinearAlgebra
 
         @test m[:, 1:2] isa Matrix
         @test m[:, [true, false, false]] isa Matrix
-        @test m[:, SOneTo(2)] isa MMatrix{2, 2, Int}
+        @test m[:, SOneTo(2)] isa SMatrix{2, 2, Int}
```

Comment:
I think this new behavior is preferable because it's more consistent. It's technically breaking, but I think the chances of this breaking the average user of `StaticArrays` are low.